mscore/prefsdialog.cpp

```diff
@@ -40,6 +40,8 @@
 #include "avsomr/avsomrlocal.h"
 #endif
 
+#define DO_NOTHING [](){}
```

Comment:
Please, *please* get rid of this macro.

Reply:
> Giving `nullptr` in place of the function is the default (only get or set) and an empty lambda means do nothing.

That's what is written in the description at the very top

Reply:
@Spire42, @igorkorsukov greenlit the use of that macro. I know that macros are best avoided (that's why I was hesitant to add it in the first place) but in this case I feel that it is appropriate (since an empty lambda does not mean much). 
Either way, it looks like the whole thing needs a good comment that explains how it works (since the difference between `[](){}` and `nullptr` is not obvious).

Reply:
Undersood.

In that case, please replace the macro with one of the following:

1. A local variable inside `PreferenceDialog::start()`:

        const auto doNothing = []() { };

2. A global variable in an unnamed namespace at the location where the macro is currently defined:

        namespace {
        const auto doNothing = []() { };
        }

(I prefer the first option.)

Reply:
`constexpr` is meant to replace macros. I would suggest `static constexpr auto doNothing = []() { }`

Reply:
I was actually going to suggest `constepxr`, but alas, `constexpr` lambdas were introduced in C++17 and we're still stuck at C++14.

Reply:
see also the comment below.

Reply:
Changed `DO_NOTHING` to `const auto doNothing = []() { };`